src/account/UpgradeableModularAccount.sol

```diff
@@ -302,6 +304,31 @@ contract UpgradeableModularAccount is
         _uninstallPlugin(plugin, manifest, pluginUninstallData, hookUnapplyData);
     }
 
+    // Should we check versions here?
+    function replacePlugin(
+        address oldPlugin,
+        address newPlugin,
+        bytes32 manifestHash,
+        bytes[] calldata hookUnapplyData
+    ) external override wrapNativeFunction {
+        address oldPluginRegistry = BasePlugin(oldPlugin).getVersionRegistry();
+        address newPluginRegistry = BasePlugin(newPlugin).getVersionRegistry();
+
+        require(oldPluginRegistry == newPluginRegistry, "Plugins use different VersionRegistries");
+
+        IVersionRegistry.Version memory oldVersion = IVersionRegistry(oldPluginRegistry).getPluginVersion(oldPlugin);
+        IVersionRegistry.Version memory newVersion = IVersionRegistry(newPluginRegistry).getPluginVersion(newPlugin);
+
+        require(
+            oldVersion.major == newVersion.major && 
+            oldVersion.minor == newVersion.minor && 
+            newVersion.patch > oldVersion.patch,
+            "New plugin is not a patch-level upgrade"
+        );
```

Comment:
Also recommend using if - revert. Also, it seems it does not allow downgrade of plugins, and is there any reason for this?